### DISCUSSION TITLE
What is FBT?

### DISCUSSION THREAD

**Discussion by ThatXliner** [astral-sh/ruff#4101] [Q&A] [answered]
I don’t understand the purpose or reason behind the [FBT](https://beta.ruff.rs/docs/rules/#flake8-boolean-trap-fbt) class of rules

**Comment by calumy** [comment, marked as answer]
This blog post might help provide more insight: https://adamj.eu/tech/2021/07/10/python-type-hints-how-to-avoid-the-boolean-trap/.

**Reply by ThatXliner** [reply]
Thanks! I think it would be very helpful to provide such a link in the documentation
